src/lib/components/ui/Button/Button.jsx

```diff
@@ -85,37 +79,56 @@ const Button = (props) => {
       >
         {props.label}
       </span>
+      {props.afterLabel && (
+        <span
+          className={`
+            ${styles.afterLabel}
+            ${props.loading ? styles.loadingIcon : ''}
+          `.trim()}
+        >
+          {props.afterLabel}
+        </span>
+      )}
+      {props.endCorner && (
+        <span className={styles.endCorner}>
+          {props.endCorner}
+        </span>
+      )}
     </button>
   );
 };
 
 Button.defaultProps = {
+  afterLabel: null,
+  beforeLabel: null,
   block: false,
   clickHandler: null,
   disabled: false,
-  icon: null,
-  iconPosition: 'before',
+  endCorner: null,
   id: undefined,
   labelVisibility: 'all',
   loading: false,
   priority: 'default',
   size: 'medium',
+  startCorner: null,
   type: 'button',
   variant: 'primary',
 };
 
 Button.propTypes = {
+  afterLabel: PropTypes.element,
+  beforeLabel: PropTypes.element,
   block: PropTypes.bool,
   clickHandler: PropTypes.func,
   disabled: PropTypes.bool,
-  icon: PropTypes.string,
-  iconPosition: PropTypes.oneOf(['before', 'after']),
+  endCorner: PropTypes.element,
   id: PropTypes.string,
   label: PropTypes.string.isRequired,
   labelVisibility: PropTypes.oneOf(['all', 'desktop', 'none']),
   loading: PropTypes.bool,
   priority: PropTypes.oneOf(['default', 'outline', 'flat']),
   size: PropTypes.oneOf(['small', 'medium', 'large']),
+  startCorner: PropTypes.element,
```

Comment:
Maybe we could rename it to `startCornerBadge`? This goes for `endCorner` as well.

Reply:
This is intentionally loose, see above.